parameter_api.cpp

```diff
@@ -9,30 +9,30 @@ static std::string trim(std::string s)
     return s.substr(first, (last-first+1));
 }
 
-void ParameterAPI::add_api_variable(std::string name, APIVariable *var) {
+void ParameterAPI::add_api_variable(const char * name, APIVariable *var) {
     variable_map_[name] = var;
     auto_complete_.add_match_string(name);
 }
 
-void ParameterAPI::add_api_variable(std::string name, const APIVariable *var) {
+void ParameterAPI::add_api_variable(const char * name, const APIVariable *var) {
     const_variable_map_[name] = var;
     auto_complete_.add_match_string(name);
 }
 
 bool ParameterAPI::set_api_variable(std::string name, std::string value) {
-    if (variable_map_.count(name))  {
-        variable_map_[name]->set(value);
+    if (variable_map_.count(name.c_str()))  {
+        variable_map_[name.c_str()]->set(value);
```

Comment:
This in particular is dicy as it stands now as technically this is a pointer to the memory backing the std::string argument which should be freed when this method returns. 

If it is the std::string_view that is showing that it doesn't own the memory and as long as it is backed by something persistent (like a string constant in static memory) then it is safe to do this. Otherwise your old approach of using a std::string as a key is safer, but requires this to make a copy of each string which takes memory.